Add modules 3 and 5

edwardyhe opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* ...

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

edwardyhe committed:
Add modules 3 and 5 drafts.

edwardyhe committed:
Merge branch 'wsilveiranz:main' into main

wsilveiranz commented:
@edwardyhe - can you check with Abode and Karan which scenario they are planning to use overall, so the connectors your are using align to it? I thought that at least for the conversational agent we were aiming to have an agent created at the end of the modules - with a few exceptions like the patterns module and the A2A module.

I can see that in the document you use RSS and MS Weather actions - but I am not sure how they would relate to the module. I thought that maybe a storage module (self contained enough) could be a good choice as it could replace some of the tools that are being mocked in the module 1.

edwardyhe committed:
Better align with style guidelines.

edwardyhe committed:
Merge branch 'main' of https://github.com/edwardyhe/logicapps-labs

ecfan commented:
@edwardyhe: Any chance you can give me edit permissions on your fork? Process will go much faster for mechanical fixes.

ecfan committed:
File name consistency update

ecfan committed:
File name consistency update

ecfan committed:
Merge branch 'main' into main

edwardyhe committed:
Update docs with verified walkthrough.

edwardyhe committed:
Add images.

edwardyhe committed:
Add images.

edwardyhe committed:
Quick pass.

ecfan commented:
@edwardyhe: I'm not sure what happened but looks like my updates to the PR got overridden. I'll need to investigate so we don't lose all the work I did.

ecfan commented:
@edwardyhe: Looks like you added image links to the original versions of the content files, rather than the updated versions with the correct file name syntax and updated content. I'm not sure how to fix this problem yet, but when I have time, I'll look into it.

edwardyhe commented on docs/logicapps-ai-course/02_build_conversational_agents/03_Connect_your_tools_with_external_services.md:
```diff
@@ -0,0 +1,217 @@
+## Cleanup
+
+- Remove demo tools if not needed.
+- If you created new connections (for example, GitHub), remove or rotate credentials per your organization's policy.
```
```suggestion
- If you created new connections (for example, GitHub), remove connections or rotate credentials per your organization's policy.
```

ecfan replied:
Resolved in latest updates.

ecfan committed:
Update with edited version and art links

Updated module title and description for clarity and consistency. Enhanced learning objectives and module content to better guide users in connecting Azure Logic Apps agents to external services.

ecfan committed:
Rename markdown file for consistency

ecfan committed:
Add updated folder name and art files

ecfan committed:
Delete folder with old file names

ecfan committed:
Revert art links back to old syntax

ecfan committed:
Fix art links

ecfan committed:
Updated art files

ecfan committed:
Updated files

ecfan committed:
Remove old file

ecfan committed:
Add relevant art to match instructions

ecfan committed:
Fix art link, clarify tool descriptions

ecfan commented on docs/logicapps-ai-course/02_build_conversational_agents/05_Add_user_context_to_your_tools.md:
```diff
@@ -1,6 +1,160 @@
+Helpful resources:
+- Microsoft identity platform: on-behalf-of flow: https://learn.microsoft.com/entra/identity-platform/v2-oauth2-on-behalf-of-flow
+- App Service authentication and authorization overview (EasyAuth): https://learn.microsoft.com/azure/app-service/overview-authentication-authorization
+- Configure Microsoft Entra ID provider (App Service authentication): https://learn.microsoft.com/azure/app-service/configure-authentication-provider-aad
```
Resolved in latest updates.

ecfan commented on docs/logicapps-ai-course/02_build_conversational_agents/05_Add_user_context_to_your_tools.md:
```diff
@@ -1,6 +1,160 @@
+
+Helpful resources:
+- Microsoft identity platform: on-behalf-of flow: https://learn.microsoft.com/entra/identity-platform/v2-oauth2-on-behalf-of-flow
+- App Service authentication and authorization overview (EasyAuth): https://learn.microsoft.com/azure/app-service/overview-authentication-authorization
```
Resolved in latest updates.

ecfan commented on docs/logicapps-ai-course/02_build_conversational_agents/05_Add_user_context_to_your_tools.md:
```diff
@@ -1,6 +1,160 @@
+Helpful resources:
+- Microsoft identity platform: on-behalf-of flow: https://learn.microsoft.com/entra/identity-platform/v2-oauth2-on-behalf-of-flow
+- App Service authentication and authorization overview (EasyAuth): https://learn.microsoft.com/azure/app-service/overview-authentication-authorization
+- Configure Microsoft Entra ID provider (App Service authentication): https://learn.microsoft.com/azure/app-service/configure-authentication-provider-aad
```
Resolved in latest updates.

ecfan commented on docs/logicapps-ai-course/02_build_conversational_agents/05_Add_user_context_to_your_tools.md:
```diff
@@ -1,6 +1,160 @@
+5. Save the configuration.
+
+Helpful links:
+- App Service authentication and authorization overview: https://learn.microsoft.com/azure/app-service/overview-authentication-authorization
```
Resolved in latest updates.

ecfan commented on docs/logicapps-ai-course/02_build_conversational_agents/05_Add_user_context_to_your_tools.md:
```diff
@@ -1,6 +1,160 @@
+
+Helpful links:
+- App Service authentication and authorization overview: https://learn.microsoft.com/azure/app-service/overview-authentication-authorization
+- Configure Microsoft Entra ID provider (App Service authentication): https://learn.microsoft.com/azure/app-service/configure-authentication-provider-aad
```
Resolved in latest updates.

ecfan committed:
Clarify instructions for cleaning up connections

ecfan committed:
Updated art and folder name

ecfan committed:
Delete old folder with old art names

ecfan committed:
Replace with edited content, including art links

ecfan committed:
Fix file name

ecfan committed:
Fix image links

ecfan committed:
Fix typo

ecfan commented on docs/logicapps-ai-course/02_build_conversational_agents/03_Connect_your_tools_with_external_services.md:
```diff
@@ -1,9 +1,222 @@
+We will keep the primary flow simple (no agent parameters and no on-behalf-of (OBO) yet). In Module 04 you will parameterize inputs, and in Module 05 you will add user context.
+
+Helpful resources:
+- Azure Logic Apps connectors overview: https://learn.microsoft.com/en-us/connectors/overview
```
Resolved in updates.

wsilveiranz committed:
Update 05-add-user-context-to-tools.md

Fixing tag typo

wsilveiranz committed:
Fix HTML line breaks in connectors and tools section
